### ISSUE TITLE
Swift 4 compatibility

### ISSUE THREAD

**Issue by jaymathew** [hackiftekhar/IQKeyboardManager#899]
Xcode 9.0 beta 3  
iOS 11 beta 3

Errors in Xcode.  I know this is beta so it might change. 
1.) Static member 'appearance' cannot be used on protocol metatype 'UIAppearance.Protocol'
2.) Declarations in extensions cannot override yet
3.) Ambiguous use of 'topMostController()'

**Comment by mekarpatx** [comment]
I have the same problem. hope it will get fixed by the Xcode.

**Comment by hackiftekhar** [comment]
@mekarpatx @jaymathew I just created swift4 branch, but I don't have Xcode9.0 beta version. Can someone make necessary changes and send me a pull request pointing to swift4 branch?

**Comment by mekarpatx** [comment]
@jaymathew do you use cocoa pods? I've found out that some files are loaded differently. 

for example: 

in my project: 
in IQToolbar.swift
`UIAppearance.appearance().barTintColor = nil` shows like this. But in GitHub, same file and same line is; 
` self.appearance().barTintColor = nil`

maybe there is a problem with the automatic migration or something else, I haven't figured out yet. 

problem could be cocoa pod or Xcode.

**Comment by jaymathew** [comment]
@mekarpatx  Yes, I do use Cocoa Pods.  I noticed the same thing you pointed out after I downloaded from your project.

**Comment by mekarpatx** [comment]
@jaymathew I've found a workaround. 
I reseted to my last working commit, didn't use pod update, don't convert to current swift, don't convert to recommended settings. 
I've selected the "swift language version" to  3.2 and  "Swift3 @obj interface" to on
hold down option key and click product, clean build folder.

after this, the project succeeded to build and run on ios11 simulator. 
in xcode9 swift 3 and 4 can work simultaneously. so no hurry to convert code. 

I'll wait to change any code until the official release, betas are full of bugs. 
hope this helps.

**Comment by jaymathew** [comment]
hey @mekarpatx  after doing what you posted is your done button showing?  I followed what you said and my done button is not showing.

**Comment by mekarpatx** [comment]
@jaymathew no, i guess its a bug with the ios 11. 
Thre are other issues regarding to done button problems but  there are no workaround that i'm aware of, yet. 
I used these steps to get rid of the errors and make a succesful build so i can test 
Best regards
